### PR TITLE
[7553] Serialize function for MsParamArray (main)

### PR DESCRIPTION
**Pull request by FifthPotato**
First attempt. Couple of things that need to be checked:
Is the type `MsParamArray**` or `MsParamArray*`? The two api calls in the api table that seem to use it both use this double-pointer. The other place where it seems like it could be used is `ExecMyRuleInp`'s member `inpParamArray`, which is a `MsParamArray*`
What is the minimum functionality that could trigger something for this? I tried doing a pep with `pep_api_exec_my_rule_post`, but couldn't seem to get any entries.

Still needs a test, once I know what to test for.

### CODE REVIEW

File: server/re/src/irods_re_serialization.cpp [hunk 1]
```diff
@@ -1199,6 +1199,69 @@ namespace irods::re_serialization
...
+                }
+                const auto* ptr = derefed_ptr->msParam[i]->inpOutBuf;
+                if (ptr) {
+                    _out[prefix + "inpOutBuf_len"] = boost::lexical_cast<std::string>(ptr->len);
+                    std::string& str = _out[prefix + "inpOutBuf_buf"];
+
+                    for (int j = 0; j < ptr->len; j++) {
+                        unsigned char c = *((unsigned char*) (ptr->buf) + j);
+
+                        if (isprint(c) || isspace(c) || isblank(c)) {
```
Comment (MartinFlores751) [inline server/re/src/irods_re_serialization.cpp:1241]: I think `isblank(..)` is covered by `isspace(...)`.

File: server/re/src/irods_re_serialization.cpp [hunk 2]
```diff
@@ -1199,6 +1199,69 @@ namespace irods::re_serialization
...
+        }
+        catch (const boost::bad_any_cast& e) {
+            return ERROR(INVALID_ANY_CAST,
+                         fmt::format("{}: failed to cast pointer to [msParamArray_t*]: {}", __func__, e.what()));
+        }
+        catch (const std::exception& e) {
+            return ERROR(
+                SYS_LIBRARY_ERROR, fmt::format("{}: failed to serialize [msParamArray_t*]: {}", __func__, e.what()));
+        }
+    }
```
Comment (MartinFlores751) [inline server/re/src/irods_re_serialization.cpp:1263]: Don't forget to comment the function name at the end of the line.

**Review by MartinFlores751 — COMMENTED** [review]
Looked over together with Derek. Minor comments on my part.

**Comment by korydraughn** [comment]
I left comments about serialization of the target type on the other PR, #7693.

**Comment by FifthPotato** [comment]
Updated with some code from the other PR. Still need to add a test.

**Comment by korydraughn** [comment]
Have you seen the test pass in isolation?
Is this ready for the full test suite?

**Comment by FifthPotato** [comment]
> Have you seen the test pass in isolation? Is this ready for the full test suite?

Test passes in isolation. If you think the code looks good, then I believe so!

**Comment by korydraughn** [comment]
Good.

I've run the unit/core tests and they all pass.

Please take care of the clang-format issue, then rebase and squash to taste, no #'s.

**Comment by FifthPotato** [comment]
Squished, rebased, clang-formatted

**Review by korydraughn — APPROVED** [review]
Pound it.

Cherry-pick to 4-3-stable.

**Comment by FifthPotato** [comment]
Octothorpe'd

**Comment by korydraughn** [comment]
Merging.

**Pull request merged**